### ISSUE TITLE
series-lines: data exceeds grid boundary when progressive rendering active and zoomed-in

### ISSUE THREAD

**Issue by dmozzoni** [apache/echarts#13197]
### Version
4.8.0

### Reproduction link
[https://codepen.io/cyric/pen/yLObeaG](https://codepen.io/cyric/pen/yLObeaG)

### Steps to reproduce
Create a chart using series-lines and force progressive rendering to be active (via "progressive" and "progressiveThreshold" keys). Then using a dataZoom component zoom in on the chart. The data in the series will often exceed the boundary of the grid. The "clip" setting doesn't seem to have any effect.
![image](https://user-images.githubusercontent.com/23322700/91375500-e1815580-e7e8-11ea-8483-963f199fcf4c.png)

### What is expected?
The data should be contained within the grid when zoomed-in

### What is actually happening?
The data exceeds the grid boundary when progressive rendering is active and zoomed-in on xAxis.

---
Set "progressive" and "progressiveThreshold" values to be smaller than the number of data points in the series to force progressive rendering to be active (disable by setting "progressive: 0"). Setting "clip' didn't seem to help.

Scatter series has similar progressive feature but doesn't seem to have the same issue when zooming.

**Comment by ngoc308** [comment]
Try to use datazoom with filterMode is 'none'.

**Comment by dmozzoni** [comment]
I just tried all the filterModes, including "none", but it didn't seem to make a difference. I also tried with the latest "5.0.0-alpha.2" version of echarts to no avail. If you fully zoom out or if progressive mode is disabled then it works correctly. Setting filterMode to "none" might have performance implications as well, and my real-world app has large datasets. I think it must be losing track of the extent of the grid.
